src/components/Stories/Comments.tsx

```diff
@@ -0,0 +1,154 @@
+import {
+    Avatar,
+    Box,
+    Button,
+    Col,
+    Input,
+    Row,
+    Text,
+    toast,
+} from '@impact-market/ui';
+import { SubmitHandler, useForm, useFormState } from 'react-hook-form';
+import { getImage } from '../../utils/images';
+import { selectCurrentUser } from '../../state/slices/auth';
+import { usePostCommentMutation } from '../../api/story';
+import { usePrismicData } from '../../libs/Prismic/components/PrismicDataProvider';
+import { useSWRConfig } from 'swr'
+import { useSelector } from 'react-redux';
+import React, { useEffect, useState } from 'react';
+import RichText from '../../libs/Prismic/components/RichText';
+import useTranslations from '../../libs/Prismic/hooks/useTranslations';
+
+const TEXT_LIMIT = 256;
+
+const Comments = (props: any) => {
+    const { comments, openNewComment, setOpenNewComment, story } = props;
+    const { t } = useTranslations();
+    const { modals } = usePrismicData();
+    const auth = useSelector(selectCurrentUser);
+    const { mutate } = useSWRConfig()
+
+    const {
+        control,
+        formState: { errors },
+        handleSubmit,
+        register,
+        reset
+    } = useForm();
+    const { isSubmitting, isSubmitSuccessful } = useFormState({
+        control
+    });
+
+    const [postComment] = usePostCommentMutation();
+
+    const onSubmit: SubmitHandler<any> = async (data) => {
+        try {      
+            //  If comment is only spaces (no characters) show error    
+            if (data?.comment?.trim().length === 0) {
+                toast.error(t('writeValidComment'))
+            } else {
+                const postRequest: any = await postComment({
```

Comment:
Can't we use swr to make this post request? Besides we're mixing both logics here, if we aren't using swr we shouldn't do the mutate below in the line 61.